library/manageiq.py

```diff
@@ -113,6 +117,28 @@ def __init__(self, module, url, user, password):
         self.changed       = False
         self.providers_url = self.api_url + '/providers'
 
+    def verify_authenticaion_validation(self, provider_id, update_time):
+        """ Returns a (success, message) tuple:
+                success: True if authentication validation passed, False otherwise
+                message: 'All Valid' if passed, authentication details otherwise
+        """
+        try:
+            epoch_update_time = int(time.mktime(time.strptime(update_time, ManageIQ.pattern)))
+            for i in range(ManageIQ.iterations):
+                result = self.client.get('{providers_url}/{id}/?attributes=authentications'.format(providers_url=self.providers_url, id=provider_id))
+                auths = [{k: auth.get(k, None) for k in ('authtype', 'status', 'status_details', 'updated_on')}
+                         for auth in result['authentications']]
+                # verify that the status reflects the current authentication validation
+                if all(int(time.mktime(time.strptime(auth['updated_on'], ManageIQ.pattern))) > epoch_update_time for auth in auths):
```

Comment:
1. can auth['update_on'] be missing (None)?
2. why `int()`?  the hardcoded format has second resolution but int doesn't add much.
3. why `mktime()`  the struct returned by strptime is perfectly comparable.
4. what if your clock (in update scenario) is slightly ahead of MIQ's clock?  IIUC if auth gets validated quickly enough, it will get stuck on time smaller than `epoch_update_time`, and you'll time out.

5. line is very long.  I'd make helper parse_time() function.  and/or move `for auth in ...` to next line.